File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-1.cs
public class Solution {
    public int FindMin(int[] nums) {
        // [3 5 2]
        // shift_l => m lowest, l lowest
        // shift_r => r lowest

        int l = 0, r = nums.Length - 1;
        int res = nums[0];
        while (l <= r) {
            int m = (r + l) / 2;

            res = Math.Min(res, nums[m]);

            if ((nums[l] < nums[m] && nums[l] < nums[r]) || (nums[m] < nums[l] && nums[m] < nums[r])) {
                r = m - 1;
            } else {
                l = m + 1;
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/house-robber/submission-1.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        
        n = len(nums)
        dp = [0] * (n + 1)
        dp[1], dp[2] = nums[0], nums[1]

        for j in range(3, n + 1):
            i = j - 1
            dp[j] = max(nums[i] + dp[j-2], nums[i] + dp[j-3])
        
        return max(dp[-1], dp[-2])

